### DISCUSSION TITLE
Questions about MotionGen

### DISCUSSION THREAD

**Discussion by Car38639** [NVlabs/curobo#171] [Software Q&A] [answered]
Greetings, thanks for the amazing work on this project! I have several questions about motion generation:

Q1:
In some cases, the plans have similar start/goal points, and we want the generated motion to be similar as well. Can MotionGen generate paths based on some previous results, such as graphs or paths?

Q2:
Is it possible to incorporate multiple intermediate points between two designated target points? These intermediate points may serve as waypoints for the path, or be utilized to optimize the trajectory that passes over them within a deviation below a certain threshold.

Thanks for your patience!

**Comment by balakumar-s** [comment, marked as answer]
Q1: We do not have a clean API for starting from previous results yet. This is something we want to look at and would be available in Summer this year.


Q2: We do not have a way to specify multiple intermediate waypoints. We do have support for:

1. A single offset way point following: https://curobo.org/source/advanced_examples/3_constrained_planning.html#grasp-approach-vector 
2. Constrain motions so that the robot takes a linear path: https://curobo.org/source/advanced_examples/3_constrained_planning.html

**Reply by Car38639** [reply]
Thanks for answering!
